src/condor_dagman/node.h

```diff
@@ -36,643 +34,467 @@
 	static int NOOP_NODE_PROCID;
-  
-    /** Constructor
-        @param jobName Name of job in dag file.  String is deep copied.
-		@param directory Directory to run the node in, "" if current
-		       directory.  String is deep copied.
-        @param cmdFile Path to condor cmd file.  String is deep copied.
-		@param submitDesc SubmitHash of all submit parameters (optional, alternative
-		    to cmdFile)
-    */
-    Node( const char* jobName,
-				const char* directory, const char* cmdFile );
-  
-    ~Node();
-
-		/** Clean up memory that's no longer needed once a node has
-			finished.  (Note that this doesn't mean that the Node object
-			is not valid -- it just cleans up some temporary memory.)
-			Also check that we got a consistent set of events for the
-			metrics.
-		*/
-	void Cleanup();
 
-	void PrefixName(const std::string &prefix);
-	inline const char* GetNodeName() const { return _nodeName; }
-	inline const char* GetDirectory() const { return _directory; }
-	inline const char* GetCmdFile() const { return _cmdFile; }
-	inline NodeID_t GetNodeID() const { return _nodeID; }
-	inline int GetRetryMax() const { return retry_max; }
-	void SetRetryMax( int new_max ) { retry_max = new_max; }
-	inline int GetRetries() const { return retries; }
-	const char* GetPreScriptName() const;
-	const char* GetPostScriptName() const;
-	const char* GetHoldScriptName() const;
-	static const char* JobTypeString() { return "HTCondor"; }
-	inline int GetProcEventsSize() const { return _gotEvents.size(); }
-	inline unsigned char GetProcEvent( int proc ) const { return _gotEvents[proc]; }
-	//If this is a factory/late materialization cluster we have to wait for a cluster
-	//remove event. Otherwise if queued nodes is 0 then all job procs are done
-	inline bool AllProcsDone() const { return !is_factory && _queuedNodeJobProcs == 0; }
+	Node(const char* jobName, const char* directory, const char* cmdFile);
+	~Node() {
+		delete _scriptPre;
+		delete _scriptPost;
+		delete _scriptHold;
+	}
 
-	bool AddScript(Script* script);
-	bool AddPreSkip( int exitCode, std::string &whynot );
+	// Cleanup node memory (Note: does not invalidate node)
+	void Cleanup();
+	// Dumb condensed node information to debug log
+	void Dump(const Dag *dag) const;
+	// double-check internal data structures for consistency
+	bool SanityCheck() const;
 
-	void SetType( NodeType type ) { _type = type; }
+	// Get this nodes unique ID number
+	inline NodeID_t GetNodeID() const { return _nodeID; }
+	// Set Type of Node
+	void SetType(NodeType type) { _type = type; }
+	// Get Type of Node
 	NodeType GetType() const { return _type; }
-	void SetNoop( bool value ) { _noop = value; }
-	bool GetNoop( void ) const { return _noop; }
-	void SetHold( bool value ) { _hold = value; }
-	bool GetHold( void ) const { return _hold; }
 
-	Script * _scriptPre;
-	Script * _scriptPost;
-	Script * _scriptHold;
+	// Check that this node can have a parent node(s)
+	bool CanAddParent(Node* parent, std::string &whynot);
```

Comment:
Can `Node *parent` be `const Node *parent` 
and also the `std::forward_list *` be const in the next line?

Also, what would it take to change this `forward_list` into a vector?